cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/InferenceConfiguration.kt

```diff
@@ -49,6 +49,12 @@ private constructor(
     /** Enables the inference of variables, such as global variables. */
     val inferVariables: Boolean,
 
+    /**
+     * A very EXPERIMENTAL feature. If this is enabled, we will try to infer return types of
+     * functions based on the context of the call it originated out of. This is disabled by default.
+     */
+    val inferReturnTypes: Boolean,
```

Comment:
I am somewhat satisfied with this feature, so we could also enable this by default

Reply:
I would agree, at least from a very uninformed perspective, to keep it disabled by default as it is described as very EXPERIMENTAL